dll/cpl/timedate/internettime.c

```diff
@@ -319,6 +351,41 @@ OnInitDialog(HWND hwnd)
     CreateNTPServerList(hwnd);
 }
 
+static VOID
+OnAutoSync(BOOL Sync)
+{
+    HKEY hKey;
+    LONG lRet;
+    LPCWSTR szAuto;
+
+    if (Sync)
+        szAuto = L"NTP\0";
+    else
+        szAuto = L"NoSync\0";
```

Comment:
Why the extra null-terminator?

Reply:
As noted in a previous comment, after changing to the example without the null terminators, when I tested this the registry got a NULL when the code was supposed to set "NTP" and got "No" when trying to set "NoSync". I changed these to add the null terminator and this made the code work correctly. I have no idea why this might be happening. Can you provide any insight? Thanks.

Reply:
That should be impossible. Your RegSetValueExW call is correct, and wcslen will return the same thing with or without the \0. There must have been something else going on.

Reply:
🤔 🤨 I would suggest retrying on a clean ROS install... But as Thomas said, this should never happen.

Reply:
Admittedly it seems to work fine now without the NULL terminator, so I have submitted another commit with the changes. Thanks.